src/hooks/preparse.ts

```diff
@@ -94,7 +94,16 @@ const hook: Hook.Preparse = async function ({ argv, options, context }) {
         if (contents === undefined) {
           return [name, undefined] satisfies [string, undefined];
         }
-        const values = ext === '.json' ? [JSON.stringify(JSON.parse(contents))] : contents?.trim().split('\n');
+        const crlf = contents.search('\r\n') !== -1;
+
+        let values;
+
+        if (crlf) {
+          values = ext === '.json' ? [JSON.stringify(JSON.parse(contents))] : contents?.trim().split('\r\n');
```

Comment:
`const values = ext === '.json' ? [JSON.stringify(JSON.parse(contents))] : contents?.trim().split( crlf ? '\r\n' : '\n')`

as an alternate

does `os.EOL` work?

Reply:
Yes it should. the fix is to detect the line ending in the file. It will looks for os.EOL('\r\n') first in the contents.